src/ast/ast.h

```diff
@@ -608,18 +608,34 @@ class Subprog : public Node {
 };
 using SubprogList = std::vector<Subprog *>;
 
+class Import : public Node {
+public:
+  Import(Diagnostics &d, std::string name, Location &&loc);
+
+  const std::string &name() const
+  {
+    return name_;
+  }
+
+private:
+  std::string name_;
```

Comment:
subprog ast has a private `name_` but all other AST classes exposes their member strings as public. This just to protect the node after creation?

Reply:
We should probably make all of these consistent. Given that Subprog and SubprogArg are already different, I can do that as a separate thing. I have a pending branch with some other consistency issues (e.g. the use of explicit constructors, other private member).